File: ios/SecretaryVaultAccess/Sources/SecretaryVaultAccess/RecoveryPhrase.swift
import Foundation

/// Normalizes user-typed BIP-39 recovery phrases before handing them to the
/// FFI: trims, lowercases, and collapses any run of whitespace to one space.
/// The canonical BIP-39 word list is all-lowercase and single-space-joined, so
/// this removes the most common copy/paste and keyboard-autocapitalization
/// noise without altering the words themselves.
public enum RecoveryPhrase {
    public static func normalize(_ raw: String) -> String {
        raw.lowercased()
            .split(whereSeparator: { $0.isWhitespace })
            .joined(separator: " ")
    }
}

File: android/vault-access/src/main/kotlin/org/secretary/browse/RecoveryPhrase.kt
package org.secretary.browse

/**
 * Normalizes a user-typed BIP-39 recovery phrase before it is handed to the FFI: lowercases,
 * splits on any whitespace run, drops empty tokens, and rejoins single-spaced. The canonical
 * BIP-39 word list is all-lowercase and single-space-joined, so this removes the most common
 * copy/paste and keyboard auto-capitalization noise without altering the words themselves.
 * Pure — no side effects. Mirror of iOS `RecoveryPhrase.normalize`.
 */
object RecoveryPhrase {
    fun normalize(raw: String): String =
        raw.lowercase()
            .split(Regex("\\s+"))
            .filter { it.isNotEmpty() }
            .joinToString(" ")
}
